### PR TITLE
HTBHF-951 Fix missing eligibility status on eligibility response

### PR DESCRIPTION
**Pull request by gethyn1**
- ~Rename `createEligibilityResponse` to `createEligibleResponse` as it will always return a response with a status of ELIGIBLE~
- Add missing eligibility status
- Write unit test for missing eligibility status bug
- Refactor test constants for integration test matching
- Write integration test for missing eligibility status bug

### CODE REVIEW

**Review by YetAnotherMatt — DISMISSED** [review]
@gethyn1 I changed the EligibilityResponseFactory so it is in line with dwp-api. I also updated ApiDocumentation so that this should be the last time swagger.yml changes without a change in the actual api.

File: api/src/test/java/uk/gov/dhsc/htbhf/hmrc/HMRCIntegrationTests.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    private RestTemplate restTemplateWithIdHeaders;
+
+    @Test
+    void shouldReturnEligibilityResponseWhenNotInDatabase() {
+        //Given
+        ResponseEntity<EligibilityResponse> hmrcEligibilityResponse = new ResponseEntity<>(anEligibilityResponse(), OK);
+        given(restTemplateWithIdHeaders.postForEntity(anyString(), any(), eq(EligibilityResponse.class))).willReturn(hmrcEligibilityResponse);
+
+        //When
+        ResponseEntity<EligibilityResponse> response = restTemplate.exchange(buildRequestEntity(aValidEligibilityRequest()), EligibilityResponse.class);
```
Comment (mikeSmith08) [inline api/src/test/java/uk/gov/dhsc/htbhf/hmrc/HMRCIntegrationTests.java]: You should use the `callService` method here too.

File: api/src/test/java/uk/gov/dhsc/htbhf/hmrc/HMRCIntegrationTests.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+    private void assertResponseCorrectWithStatusOnly(ResponseEntity<EligibilityResponse> response, EligibilityStatus status) {
+        EligibilityResponse expectedResponse = EligibilityResponse.builder()
+                .eligibilityStatus(status)
+                .build();
+        assertResponseCorrect(response, expectedResponse);
+    }
+
+    private void assertResponseCorrect(ResponseEntity<EligibilityResponse> response, EligibilityResponse expectedResponse) {
+        AssertionsForInterfaceTypes.assertThat(response.getStatusCode()).isEqualTo(OK);
```
Comment (mikeSmith08) [inline api/src/test/java/uk/gov/dhsc/htbhf/hmrc/HMRCIntegrationTests.java]: That's the wrong static import, plus for below too.

**Review by mikeSmith08 — CHANGES_REQUESTED** [review]

File: api/src/test/java/uk/gov/dhsc/htbhf/hmrc/HMRCIntegrationTests.java (hunk 2 above)
Comment (gethyn1) [reply api/src/test/java/uk/gov/dhsc/htbhf/hmrc/HMRCIntegrationTests.java]: Good spot, hopefully correct now.

File: api/src/test/java/uk/gov/dhsc/htbhf/hmrc/HMRCIntegrationTests.java (hunk 1 above)
Comment (gethyn1) [reply api/src/test/java/uk/gov/dhsc/htbhf/hmrc/HMRCIntegrationTests.java]: Refactored to use `callService()`

**Review by YetAnotherMatt — APPROVED** [review]

**Review by mikeSmith08 — APPROVED** [review]

**Pull request merged**